common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -293,19 +293,15 @@ def publish_selected_children_events(cls, block_keys, format_block_keys, publish
 
     def selected_children(self):
         """
-        Returns a set() of block_ids indicating which of the possible children
+        Returns a list() of block_ids indicating which of the possible children
         have been selected to display to the current user.
 
         This reads and updates the "selected" field, which has user_state scope.
 
-        Note: self.selected and the return value contain block_ids. To get
+        Note: the return value (self.selected) contains block_ids. To get
         actual BlockUsageLocators, it is necessary to use self.children,
         because the block_ids alone do not specify the block type.
         """
-        if hasattr(self, "_selected_set"):
-            # Already done:
-            return self._selected_set  # pylint: disable=access-member-before-definition
```

Comment:
@bradenmacdonald, when I tested it, the memoization didn't really work and the condition was failing on every (re)load of the page containing the XBlock and the `make_selection` method was getting called. @kaizoku also saw the same behaviour and confirmed this.

The `make_selection` method call essentially behaves like a no-op whenever the number of blocks to be selected or the eligible blocks to be selected do not change. If either of those change, the memoization wouldn't help because the selection has to be made again.

Reply:
@lgp171188 The memoization was only for a single request-response cycle, so it was expected that if you reload the page, it would call `make_selection` again. The main concern is just to make sure that a single load of the page results in no more than one call to `make_selection`.

Reply:
> The memoization was only for a single request-response cycle,

@bradenmacdonald, during my testing, the `selected_children` method (which had the memoization and conditionally called the `make_selection` method) was called only once and that was the time of loading the page. So I am not able to understand how memoization will help. Did I miss something crucial to understanding this? Can you explain?

Reply:
@lgp171188 In that case, it's fine :) I thought that there might be some circumstances where it got called more than once per page load, but if that's not the case, then there's no need to memoize it.